core/src/main/java/com/sequenceiq/cloudbreak/core/flow2/externaldatabase/ExternalDatabaseService.java

```diff
@@ -216,6 +219,18 @@ public void rotateDatabaseSecret(String databaseServerCrn, RedbeamsSecretType se
         }
     }
 
+    public void preValidateDatabaseSecretRotation(String databaseServerCrn) {
+        if (StringUtils.isEmpty(databaseServerCrn)) {
+            throw new SecretRotationException("No database server crn found, rotation is not possible.", null);
```

Comment:
Can we have one of these instead of passing in null:
1. a rotation step as Validate 
2. or have a constructor without a rotationStep
3. or a custom exception for RotationValidationException
?

Reply:
I see an approve, but no resolution for this, Shall we wait for the answer/fix, or should it be merged?

Reply:
I will fix this in a later PR.